### PR TITLE
Dịch "numerical-stability-and-init_vn" - Phần 2

### LINKED ISSUE #980: Dịch "numerical-stability-and-init_vn" - Phần 2
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md#L67-L138

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tamnguyenvan**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #980

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 1]
```diff
@@ -124,7 +131,10 @@ For the scale that we picked (the choice of the variance $\sigma^2=1$), the matr
 If this were to happen to us with a deep network, we would have no realistic chance of getting a gradient descent optimizer to converge.
 -->
 
-*dịch đoạn phía trên*
+Một vấn đề đối lập, bùng nổ gradient, cũng có thể gây phiền toán không kém.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Một vấn đề đối lập, bùng nổ gradient, cũng có thể gây phiền toái không kém.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 2]
```diff
@@ -79,7 +79,10 @@ Since early artificial neural networks were inspired by biological neural networ
 Let's take a closer look at the function to see why picking it might be problematic vis-a-vis vanishing gradients.
 -->
 
-*dịch đoạn phía trên*
+Một thủ phạm chính gây ra vấn đề tiêu biến gradient là việc lựa chọn hàm kích hoạt $\sigma$, là hàm được đặt xen giữa các toán hạng tuyến tính tại mỗi tầng.
```
Comment (lkhphuc) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Một thủ phạm chính gây ra vấn đề tiêu biến gradient là hàm kích hoạt $\sigma$ được chọn để đặt xen giữa các toán hạng tuyến tính tại mỗi tầng.
```

**Comment by tamnguyenvan** [comment]
@lkhphuc Cảm ơn bạn đã review

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 3]
```diff
@@ -79,7 +79,10 @@ Since early artificial neural networks were inspired by biological neural networ
 Let's take a closer look at the function to see why picking it might be problematic vis-a-vis vanishing gradients.
 -->
 
-*dịch đoạn phía trên*
+Một thủ phạm chính gây ra vấn đề tiêu biến gradient là hàm kích hoạt $\sigma$ được chọn để đặt xen giữa các toán hạng tuyến tính tại mỗi tầng.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]: mình tra `toán hạng` thì có nghĩa giống `input`, `operation` là `phép toán` thì đúng hơn
```suggestion
Một thủ phạm chính gây ra vấn đề tiêu biến gradient là hàm kích hoạt $\sigma$ được chọn để đặt xen giữa các phép toán tuyến tính tại mỗi tầng.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 4]
```diff
@@ -79,7 +79,10 @@ Since early artificial neural networks were inspired by biological neural networ
...
+Trước đây, hàm kích hoạt sigmoid $(1 + \exp(-x))$ (đã giới thiệu trong :numref:`sec_mlp`) là lựa chọn phổ biến bởi nó hoạt động giống với một hàm lấy ngưỡng.
+Cũng bởi các mạng nơ-ron nhân tạo thời kỳ đầu lấy cảm hứng từ mạng nơ-ron sinh học, ý tưởng các nơ-ron được kích hoạt hoặc không bị kích hoạt (nơ-ron sinh học không bị kích hoạt một phần) có vẻ như rất hấp dẫn.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Cũng bởi các mạng nơ-ron nhân tạo thời kỳ đầu lấy cảm hứng từ mạng nơ-ron sinh học, ý tưởng các nơ-ron được kích hoạt hoặc không bị kích hoạt (nơ-ron sinh học không bị kích hoạt một phần) có vẻ rất hấp dẫn.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 5]
```diff
@@ -79,7 +79,10 @@ Since early artificial neural networks were inspired by biological neural networ
...
+Hãy xem xét chi tiết hơn để thấy tại sao sử dụng hàm sigmoid có thể sinh ra vấn đề liên quan tới hiện tượng tiêu biến gradient.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Hãy xem xét chi tiết hơn để thấy tại sao việc sử dụng hàm sigmoid có thể gây ra vấn đề liên quan tới hiện tượng tiêu biến gradient.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 6]
```diff
@@ -105,7 +108,11 @@ Before ReLUs ($\max(0, x)$) were proposed as an alternative to squashing functio
 As a consequence, ReLUs have become the default choice when designing activation functions in deep networks.
 -->
 
-*dịch đoạn phía trên*
+Như ta có thể thấy, gradient của hàm sigmoid tiêu biến khi đầu vào của nó quá lớn hoặc quá nhỏ.
+Hơn nữa, khi chúng ta thực hiện lan truyền ngược, dùng quy tắc dây chuyền, trừ khi giá trị nằm trong vùng Goldilocks, tại đó đầu vào của hầu hết các hàm sigmoid nằm trong khoảng, ví dụ $[-4, 4]$, gradient của tất cả các phép nhân đều có thể bị tiêu biến.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Hơn nữa, khi chúng ta thực hiện lan truyền ngược, dùng quy tắc dây chuyền, trừ khi giá trị nằm trong vùng Goldilocks, tại đó đầu vào của hầu hết các hàm sigmoid nằm trong khoảng, ví dụ $[-4, 4]$, gradient của cả phép nhân có thể bị tiêu biến.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 7]
```diff
@@ -105,7 +108,11 @@ Before ReLUs ($\max(0, x)$) were proposed as an alternative to squashing functio
...
+Khi chúng ta có nhiều tầng, trừ khi ta cực kỳ cẩn trọng, nhiều khả năng ta sẽ thấy gradient bị mất tại *một vài* tầng nào đó.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]: mình thấy `mất` chưa diễn tả đủ từ `cut off`
```suggestion
Khi chúng ta có nhiều tầng, trừ khi ta cực kỳ cẩn trọng, nhiều khả năng ta sẽ thấy luồng gradient bị ngắt tại *một* tầng nào đó.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 8]
```diff
@@ -105,7 +108,11 @@ Before ReLUs ($\max(0, x)$) were proposed as an alternative to squashing functio
...
+Trước khi hàm ReLU ($\max(0, x)$) được đề xuất để thay thế các hàm kiểu sigmoid, vấn đề này đã từng gây nhiều khó khăn cho quá trình huấn luyện mạng học sâu.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]: `squashing function` trước có người dịch thế này thì phải
```suggestion
Trước khi hàm ReLU ($\max(0, x)$) được đề xuất để thay thế các hàm nén, vấn đề này đã từng gây nhiều khó khăn cho quá trình huấn luyện mạng nơ-ron sâu.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 9]
```diff
@@ -105,7 +108,11 @@ Before ReLUs ($\max(0, x)$) were proposed as an alternative to squashing functio
...
+Kết quả là, ReLU dần trở thành lựa chọn mặc định khi thiết kế các hàm kích hoạt trong mạng học sâu.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Kết quả là, ReLU dần trở thành lựa chọn mặc định khi thiết kế các hàm kích hoạt trong mạng nơ-ron sâu.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 10]
```diff
@@ -124,7 +131,10 @@ For the scale that we picked (the choice of the variance $\sigma^2=1$), the matr
 If this were to happen to us with a deep network, we would have no realistic chance of getting a gradient descent optimizer to converge.
 -->
 
-*dịch đoạn phía trên*
+Một vấn đề đối lập, bùng nổ gradient, cũng có thể gây phiền toái không kém.
+Để chứng minh rõ hơn điều này, chúng ta lấy mẫu $100$ ma trận ngẫu nhiên Gaussian và nhân chúng với một vài ma trận khởi tạo.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Để biểu diễn việc này rõ hơn, chúng ta lấy $100$ ma trận ngẫu nhiên Gaussian và nhân chúng với một ma trận khởi tạo.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 11]
```diff
@@ -124,7 +131,10 @@ For the scale that we picked (the choice of the variance $\sigma^2=1$), the matr
...
+Với khoảng giá trị mà ta đã chọn (phương sai $\sigma^2=1$), tích các ma trận bị bùng nổ số học.
+Nếu điều này xảy ra trong các mạng học sâu, chúng ta sẽ không thể làm các bộ tối ưu dựa trên giảm gradient hội tụ được.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
Nếu điều này xảy ra trong các mạng học sâu, các bộ tối ưu dựa trên hạ gradient sẽ không thể hội tụ được.
```

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md [hunk 12]
```diff
@@ -352,7 +362,8 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 *
 
 <!-- Phần 2 -->
-*
+* Nguyễn Văn Tâm
+* Phúc Lê
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]:
```suggestion
* Lê Khắc Hồng Phúc
```
I can't say anything... (**[Danh sách Contributors](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**)

File: chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md (hunk 3 above)
Comment (lkhphuc) [reply chapter_multilayer-perceptrons/numerical-stability-and-init_vn.md]: `operator` cũng có thể dịch là `toán tử ` (https://vi.wikipedia.org/wiki/To%C3%A1n_t%E1%BB%AD)

**Comment by duythanhvn** [comment]
Done thôi em ơi.

**Comment by tamnguyenvan** [comment]
Sorry mọi người, mấy này mình hơi bận chút :D
